inc/vcf/summary_report_writer.hpp

```diff
@@ -24,88 +24,50 @@ namespace ebi
 {
   namespace vcf
   {
+
     /**
-     * Class that tells whether an error should be written or skipped.
+     * Stores the count and first line of appearance of an error message
+     */
+    struct ErrorSummary
+    {
+      size_t appearances;
+      size_t first_appearance_line;
+    };
+
+    /**
+     * Class that ensures similar kind of errors are reported only once.
      *
-     * The intended algorithm is that all errors will be printed the first time they appear, but some of them won't
+     * The intended algorithm is that errors and warnings will be printed only the first time they appear, and won't
      * be printed again.
      *
-     * To differentiate the Error dynamic type, this class implements ErrorVisitor. As the visitor interface returns
-     * `void`, we have to store the decision in the class' state, in the variable `skip`.
+     * The summary diplays the count(number of times it appears) of the error, and the line number of it's first
+     * occurrence. We distinguish between different types of errors based on their simple error message (which contains
+     * no details). The `error_order` basically maintains the order in which these errors appaer for the first time.
      *
      * Use this class like this:
      * ~~~
-     * if (summary.should_write_report(error)) {
-     *     std::cout << error.what() << " (warning)" << std::endl;
-     * }
+     * summary.add_to_summary(error.message, error.line);
```

Comment:
now that the interface of this class is much simpler, we can remove this comment on how to use this class